.circleci/config.yml

```diff
@@ -0,0 +1,57 @@
+step-restore-cache: &step-restore-cache
+  restore_cache:
+    keys:
+      - v1-dependencies-{{ arch }}-{{ checksum "yarn.lock" }}
+      - v1-dependencies-{{ arch }}
+
+steps-test: &steps-test
+  steps:
+    - checkout
+    - *step-restore-cache
+    - run: yarn --frozen-lockfile
+    - save_cache:
+        paths:
+          - node_modules
+        key: v1-dependencies-{{ arch }}-{{ checksum "yarn.lock" }}
+    - run: yarn test
+
+version: 2.1
+jobs:
+  test-linux-10:
+    docker:
+      - image: circleci/node:10
+    <<: *steps-test
+  test-linux-12:
+    docker:
+      - image: circleci/node:12
+    <<: *steps-test
+  test-linux-14:
+    docker:
+      - image: circleci/node:14
+    <<: *steps-test
+
+  release:
+    docker:
+      - image: circleci/node:14.15
+    steps:
+      - checkout
+      - *step-restore-cache
+      - run: yarn --frozen-lockfile
+      - run: npx semantic-release@17.4.5
+workflows:
+  version: 2
+  test_and_release:
+    # Run test jobs first, release only when all the test jobs are successful
+    jobs:
+      - test-linux-10
+      - test-linux-12
+      - test-linux-14
+      - release:
+          requires:
+            - test-linux-10
+            - test-linux-12
+            - test-linux-14
+          filters:
+            branches:
+              only:
+                - master
```

Comment:
Once we have the CI pipeline up and running, it would awesome to port the default branch over to main; I did a quick sweep of the repo and this PR has the only places where "master" is hardcoded 🙂 But we can make sure this is working first!